internal/controller/postgrescluster/instance_test.go

```diff
@@ -2271,3 +2274,201 @@ func fakeUpgradeCluster(clusterName, namespace, clusterUID string) *v1beta1.Post
+
+	r := &Reconciler{
+		Client: cc,
+		Owner:  client.FieldOwner(t.Name()),
+	}
+
+	foundPDB := func(
+		cluster *v1beta1.PostgresCluster,
+		spec *v1beta1.PostgresInstanceSetSpec,
+	) bool {
+		got := &policyv1beta1.PodDisruptionBudget{}
+		err := r.Client.Get(ctx,
+			naming.AsObjectKey(naming.InstanceSet(cluster, spec)),
+			got)
+		return !apierrors.IsNotFound(err)
+
+	}
+
+	ns := &corev1.Namespace{}
+	ns.GenerateName = "postgres-operator-test-"
+	ns.Labels = labels.Set{"postgres-operator-test": t.Name()}
+	assert.NilError(t, cc.Create(ctx, ns))
+	t.Cleanup(func() { assert.Check(t, cc.Delete(ctx, ns)) })
+
+	t.Run("empty", func(t *testing.T) {
+		cluster := &v1beta1.PostgresCluster{}
+		spec := &v1beta1.PostgresInstanceSetSpec{}
+
+		assert.Equal(t, r.reconcileInstanceSetPodDisruptionBudget(ctx, cluster, spec).Error(),
+			"Replicas should be defined")
+	})
+
+	t.Run("not created", func(t *testing.T) {
+		cluster := testCluster()
+		cluster.Namespace = ns.Name
+		spec := &cluster.Spec.InstanceSets[0]
+		spec.MinAvailable = &intstr.IntOrString{IntVal: int32(0)}
+		assert.NilError(t, r.reconcileInstanceSetPodDisruptionBudget(ctx, cluster, spec))
+		assert.Assert(t, !foundPDB(cluster, spec))
+	})
+
+	t.Run("int created", func(t *testing.T) {
+		cluster := testCluster()
+		cluster.Namespace = ns.Name
+		spec := &cluster.Spec.InstanceSets[0]
+		spec.MinAvailable = &intstr.IntOrString{IntVal: int32(1)}
+
+		assert.NilError(t, r.Client.Create(ctx, cluster))
+		t.Cleanup(func() { assert.Check(t, r.Client.Delete(ctx, cluster)) })
+
+		assert.NilError(t, r.reconcileInstanceSetPodDisruptionBudget(ctx, cluster, spec))
+		assert.Assert(t, foundPDB(cluster, spec))
+
+		t.Run("delete pdb", func(t *testing.T) {
+			spec.MinAvailable = &intstr.IntOrString{IntVal: int32(0)}
+			err := r.reconcileInstanceSetPodDisruptionBudget(ctx, cluster, spec)
+			if apierrors.IsConflict(err) {
+				// When running in an existing environment another controller will sometimes update
+				// the object. This leads to an error where the ResourceVersion of the object does
+				// not match what we expect. When we run into this conflict, try to reconcile the
+				// object again.
+				err = r.reconcileInstanceSetPodDisruptionBudget(ctx, cluster, spec)
+			}
+			assert.NilError(t, err, errors.Unwrap(err))
+			assert.Assert(t, !foundPDB(cluster, spec))
+		})
+	})
+
+	t.Run("str created", func(t *testing.T) {
+		cluster := testCluster()
+		cluster.Namespace = ns.Name
+		spec := &cluster.Spec.InstanceSets[0]
+		spec.MinAvailable = &intstr.IntOrString{
+			Type:   1,
+			StrVal: "50%",
+		}
+
+		assert.NilError(t, r.Client.Create(ctx, cluster))
+		t.Cleanup(func() { assert.Check(t, r.Client.Delete(ctx, cluster)) })
+
+		assert.NilError(t, r.reconcileInstanceSetPodDisruptionBudget(ctx, cluster, spec))
+		assert.Assert(t, foundPDB(cluster, spec))
+
+		t.Run("delete pdb", func(t *testing.T) {
+			spec.MinAvailable = &intstr.IntOrString{IntVal: int32(0)}
```

Comment:
May be silly of me, but I'd like to see a string value of "0%" as part of the delete here -- I mean, it was created with a string, why not delete it with a string too?